tests/jest/view/plugin/search/search.test.tsx

```diff
@@ -37,60 +21,87 @@ describe('Search plugin', () => {
   });
 
   it('should render the search box', async () => {
-    const mock = jest.spyOn(SearchActions.prototype, 'search');
+    const mock = jest.spyOn(SearchActions, 'SearchKeyword');
+
+    config.update({
+      search: {
+        keyword: 'boo',
+      },
+    });
 
     const search = mount(
-      <configContext.Provider value={config}>
-        <Search plugin={plugin} enabled={true} keyword={'boo'} />
-      </configContext.Provider>,
+      <ConfigContext.Provider value={config}>
+        <Search />
+      </ConfigContext.Provider>,
     );
 
     expect(mock).toBeCalledWith('boo');
     expect(search.html()).toMatchSnapshot();
   });
 
   it('should not call search if keyword is undefined', async () => {
-    const mock = jest.spyOn(SearchActions.prototype, 'search');
+    const mock = jest.spyOn(SearchActions, 'SearchKeyword');
+
+    config.update({
+      search: true,
+    });
 
     mount(
-      <configContext.Provider value={config}>
-        <Search plugin={plugin} enabled={true} />
-      </configContext.Provider>,
+      <ConfigContext.Provider value={config}>
+        <Search />
+      </ConfigContext.Provider>,
     );
 
     expect(mock).not.toBeCalled();
   });
 
   it('should call search action after input change', async () => {
-    const mock = jest.spyOn(SearchActions.prototype, 'search');
+    const mock = jest.spyOn(SearchActions, 'SearchKeyword');
+
+    config.update({
+      search: true,
+    });
 
     const wrapper = mount(
-      <configContext.Provider value={config}>
-        <Search plugin={plugin} enabled={true} />
-      </configContext.Provider>,
+      <ConfigContext.Provider value={config}>
+        <Search />
+      </ConfigContext.Provider>,
     );
 
-    // https://github.com/preactjs/enzyme-adapter-preact-pure/issues/45
     const input = wrapper.find('input');
-    input.getDOMNode<HTMLInputElement>().value = '123';
+    const onInput = input.props().onInput;
 
-    input.simulate('input');
+    await act(() => {
+      const htmlInputElement = document.createElement('input');
+      htmlInputElement.value = '123';
+      onInput({ target: htmlInputElement });
+    });
+
+    wrapper.update();
+
+    await flushPromises();
 
-    expect(mock).toBeCalledWith('123');
+    return new Promise<void>((resolve) => {
+      // TODO: can we fix this and remove the setTimeout?
+      setTimeout(() => {
```

Comment:
I know with Testing library you have access to a 'waitfor' function that basically tries your assertion for X seconds.

Maybe enzyme has something similar?
Or we could rig our own?

Reply:
I'm planning on migrating to the React Testing Library which could potentially help us remove these setTimeouts.